### PR TITLE
feat: add NodeTypes to add and remove elements from a list

### PR DESCRIPTION
**Pull request by Le-Pepp**

### CODE REVIEW

File: basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/AddElementToListNodeType.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    public Map<String, CompletableFuture<DataBox<?>>> compute(Map<String, DataBox<?>> inputs, Map<String, DataBox<?>> settings, Map<String, DataType<?>> inputTypes) {
+        if (inputTypes.get("element").listOf() != inputTypes.get("list")) {
+            throw new IllegalArgumentException("The Element's Type does not match that of the List");
+        }
+        return Map.of(
+            "output", CompletableFuture.completedFuture(this.getGenericListBox(
+                Stream.concat(
+                    DataBox.get(inputs, "list", ListDataType.GENERIC_LIST).orElse(List.of()).stream(),
+                    Stream.of(DataBox.get(inputs, "element", DataType.ANY).orElseThrow()))
+                .collect(Collectors.toList()),
```
Comment (williambl) [inline basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/AddElementToListNodeType.java:49]: Please just copy the list and add the element this is mildly confusing to read and is probably many times slower.

File: basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/RemoveElementFromListNodeType.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            } catch (ClassCastException e) {
+                return Map.of(
+                    "output", CompletableFuture.completedFuture(this.getGenericListBox(inputList, listType))
+                );
+            }
+        }
+
+        if (mode == Mode.INDEX) index = (int) (double) DataBox.get(inputs, "index", DataType.NUMBER).orElseThrow();
+
+        System.out.println(index);
```
Comment (williambl) [inline basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/RemoveElementFromListNodeType.java:74]: Please remove.

File: basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/RemoveElementFromListNodeType.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+        if (index == -1) return Map.of("output", CompletableFuture.completedFuture(this.getGenericListBox(inputList, listType)));
+   
+        return Map.of(
+            "output", CompletableFuture.completedFuture(this.getGenericListBox(
+                Stream.concat(
+                    inputList.subList(0, index).stream(),
+                    inputList.subList(index + 1, inputList.size()).stream())
+                    .collect(Collectors.toList()),
```
Comment (williambl) [inline basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/RemoveElementFromListNodeType.java:83]: Please just copy the list and remove the element this is _really_ confusing to read and is probably many times slower.

**Review by williambl — APPROVED** [review]

**Comment by ShaksterNano** [comment]
Merge!!!1!1!1!1!1

**Comment by fenn7** [comment]
FIX CONFLICTS !!!!!

**Pull request merged**